include/arch/x86/pagetable.h

```diff
@@ -272,7 +272,6 @@ static inline pdpe_t *get_pdpe(const void *va) {
 static inline void set_pgentry(pgentry_t *e, mfn_t mfn, unsigned long flags) {
     *e = pgentry_from_mfn(mfn, flags);
     barrier();
-    flush_tlb();
```

Comment:
I guess you removed the TLB flushing here and pushed the responsibility to the callers just to be able to explicitly omit it in `get_pgentry_mfn()`? If so, is there guaranteed to be no problem with stale paging structure cache entries? (No TLB entry, as it's a !p -> p change, but still the paging structure cache might be prepoupalated?)

Reply:
Yes, that was one of the reasons. Another was to make it less strict in order to allow various combinations of direct usage. So far the explicit mappings in the `_vmap()` are guarded with the TLB invalidation, so the core should be fine (the `get_pgentry_mfn()`'s change should be covered by the invalidation in `tmp_map_mfn()`).
Whenever a test needs to do something less strict, it can use the interface directly.